content/immutable-owner.md

```diff
@@ -0,0 +1,393 @@
+
+### 1. Setup Environment
+
+To get started, create an empty directory named `immutable-owner` and navigate to it. We'll be initializing a brand new project. Run `npm init` and follow through the prompts.
+
+Next, we'll need to add our dependencies. Run the following to install the required packages:
+```bash
+npm i @solana-developers/helpers @solana/spl-token @solana/web3.js esrun dotenv typescript
+```
+
+Create a directory named `src`. In this directory, create a file named `index.ts`. This is where we will run checks against the rules of this extension. Paste the following code in `index.ts`:
+
+```ts
+import { AuthorityType, TOKEN_2022_PROGRAM_ID, createMint, setAuthority } from "@solana/spl-token";
+import { Connection, Keypair, LAMPORTS_PER_SOL, PublicKey } from "@solana/web3.js";
+import { initializeKeypair, makeKeypairs } from "@solana-developers/helpers";
+// import { createTokenAccountWithImmutableOwner } from "./token-helper"; // We will uncomment later
+
+const connection = new Connection("http://127.0.0.1:8899", 'confirmed');
+const payer = await initializeKeypair(connection);
+
+const [otherOwner, mintKeypair, ourTokenAccountKeypair] = makeKeypairs(3)
+const ourTokenAccount = ourTokenAccountKeypair.publicKey;
+
+// CREATE MINT
+
+// CREATE TEST TOKEN ACCOUNTS: Create explicitly with immutable owner instructions
+
+// CREATE TEST TOKEN ACCOUNTS: Create an associated token account with default immutable owner
+
+// TEST TRANSFER ATTEMPT ON IMMUTABLE ACCOUNT
+
+// TEST TRANSFER ATTEMPT ON ASSOCIATED IMMUTABLE ACCOUNT
+```
+
+### 2. Run validator node
+
+For the sake of this guide, we'll be running our own validator node.
+
+In a separate terminal, run the following command: `solana-test-validator`. This will run the node and also log out some keys and values. The value we need to retrieve and use in our connection is the JSON RPC URL, which in this case is `http://127.0.0.1:8899`. We then use that in the connection to specify to use of the local RPC URL.
+
+```tsx
+const connection = new Connection("http://127.0.0.1:8899", "confirmed");
+```
+
+Alternatively, if you’d like to use testnet or devnet, import the `clusterApiUrl` from `@solana/web3.js` and pass it to the connection as such:
+
+```tsx
+const connection = new Connection(clusterApiUrl('devnet'), 'confirmed');
+```
+
+If you decide to use devnet, and have issues with airdropping SOL. Feel free to add the `keypairPath` parameter to `initializeKeypair`. You can get this by running `solana config get` in your terminal. And then go to [faucet.solana.com](https://faucet.solana.com/) and airdrop some SOL to your address. You can get your address by running `solana address` in your terminal.
+
+### 3. Helpers
+
+When we pasted the `index.ts` code from earlier, we added the following helpers:
+
+- `initializeKeypair`: This function creates the keypair for the `payer` and also airdrops 2 testnet SOL to it
```

Comment:
We have `airdropIfRequired()` for this, let's use that instead of adding another helper that is specific to this lab.

Reply:
`initializeKeypair` is also in the solana dev repo